sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/unresolved.scala

```diff
@@ -244,7 +244,8 @@ case class UnresolvedGenerator(name: FunctionIdentifier, children: Seq[Expressio
 case class UnresolvedFunction(
     name: FunctionIdentifier,
     children: Seq[Expression],
-    isDistinct: Boolean)
+    isDistinct: Boolean,
+    filter: Option[Expression] = None)
```

Comment:
shall we treat `filter` as child as well? e.g.
```
case class UnresolvedFunction(... inputs: Seq[Expression], ... filter: Option[Expression] = None) {
  def children = inputs ++ filter
}
```

Reply:
Yes, we can do this.